src/safety/safety_check_v2.py

```diff
@@ -0,0 +1,110 @@
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import json
+import sys
+import argparse
+from datetime import datetime
+import pkg_resources
+import safety as Safety
+from safety import safety
+
+
+def main():
+    """Return json.dumps() string of below list structure.
+    [
+        {
+            "package": "package",
+            "affected": "version_spec",
+            "installed": "version",
+            "vulnerabilities": [
+                {
+                    "vid": "safety_vulnerability_id",
+                    "advisory": "description of the issue"
+                },
+                ...
+            ]
+        }
+        ...
+    ]
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--key",
+        required=True,
+    )
+    args = parser.parse_args()
+    safety_key = args.key
+
+    if Safety.__version__ == "1.8.7":
+        proxy_dictionary = {}
+    else:
+        from safety.util import get_proxy_dict
+        proxy_dictionary = get_proxy_dict("http", None, 80)
+
+    # Get safety result
+    packages = [
+        d for d in pkg_resources.working_set if d.key not in {"python", "wsgiref", "argparse"}
+    ]
+    # return: Vulnerability(namedtuple("Vulnerability", ["name", "spec", "version", "advisory", "vuln_id", "cvssv2", "cvssv3"]))
+    vulns = safety.check(
+        packages=packages,
+        key=safety_key,
+        db_mirror="",
+        cached=True,
+        ignore_ids=[],
+        proxy=proxy_dictionary,
+    )
+
+    # Generate output
+    vulns_dict = {}
+    vulns_list = []
+    # Report for unsafe packages
+    for v in vulns:
+        package = v.name
+        affected = v.spec
+        installed = v.version
+        advisory = v.advisory
+        vid = v.vuln_id
+        if package not in vulns_dict:
+            vulns_dict[package] = {
+                "package": package,
+                "affected": affected,
+                "installed": installed,
+                "vulnerabilities": [{"vid": vid, "advisory": advisory}],
+            }
+        else:
+            vulns_dict[package]["vulnerabilities"].append({"vid": vid, "advisory": advisory})
+
+    # Report for safe packages
+    timestamp = datetime.now().strftime("%d%m%Y")
+    for pkg in packages:
+        if pkg.key not in vulns_dict:
+            vulns_dict[pkg.key] = {
+                "package": pkg.key,
+                "affected": f"No known vulnerability found, PASSED_SAFETY_CHECK on {timestamp}.",
+                "installed": pkg.version,
+                "vulnerabilities": [{"vid": "N/A", "advisory": "N/A"}],
+            }
+        
+    for (k, v) in vulns_dict.items():
+        vulns_list.append(v)
+
+    print(json.dumps(vulns_list))
```

Comment:
nit: let's use python logging library to print the json obj

Reply:
as long as it output the json text without anything else, then it works.